virtwho/register.py

```diff
@@ -443,6 +443,302 @@ def associate(self, host_name, guest_uuid):
         return False
 
 
-class SatelliteCLI:
-    def __init__(self):
-        pass
+class Satellite:
+
+    def __init__(self, org=None, activation_key=None):
+        """
+        Using hammer command to set satellite, handle organization and
+        activation key, attach/remove subscription for host.
+        Using api to check the host-to-guest associations.
+        :param org: organization label, use the default_org configured
+            in virtwho.ini as default.
+        :param activation_key: activation key name, use the configure
+            in virtwho.ini as default.
+        """
+        register = get_register_handler('satellite')
+        self.org = org or register.default_org
+        self.activation_key = activation_key or register.activation_key
+        self.ssh = SSHConnect(host=register.server,
+                              user=register.ssh_username,
+                              pwd=register.ssh_password)
+        self.org_id = self.org_id()
+        self.api = f'https://{register.server}'
+        self.auth = (register.username, register.password)
+
+    def org_id(self, org=None):
+        """
+        Get the organization id by organization label.
+        :param org: organization label, use the org when instantiate
+        the class as default
+        :return: organization id
+        """
+        org = org or self.org
+        ret, output = self.ssh.runcmd(f'hammer organization info '
+                                      f'--label "{org}" '
+                                      f'--fields Id')
+        if ret == 0 and 'Id' in output:
+            org_id = output.split(':')[1].strip()
+            return org_id
```

Comment:
There is a good function from robottelo about the hammer output:
https://github.com/SatelliteQE/robottelo/blob/master/robottelo/cli/hammer.py#L197 
it will parse the hammer output and returns a dict mapping the values.

Reply:
Referred to the hammer.py in robottelo, I found we can get output with different format ```'base', 'table', 'silent', 'csv', 'yaml', 'json'```. So I updated the code again to use ```hammer --output=json``` as the default to get results, which is more clear.

If there will be requirements to analyze result like like robottelo, will do it in the future.